mission_control/navigator_missions/nav_missions_lib/mission_planner_no_markers.py

```diff
@@ -0,0 +1,291 @@
+            fprint(err, msg_color="red", title="{} MISSION ERROR: ".format(self.current_mission_name))
+        yield self.current_mission.safe_exit(self.navigator, err, self, self.module)
+        self._mission_complete(self.current_mission)
+        self.current_mission = None
+
+    def _end_mission(self, result):
+        self.pub_msn_info.publish(String("Ending Mission {}".format(self.current_mission_name)))
+        fprint(str(result) + " TIME: " + str((self.nh.get_time() - self.current_mission_start_time).to_sec()),
+               msg_color="green", title="{} MISSION COMPLETE: ".format(self.current_mission_name))
+        self.running_mission = False
+        self.current_mission_name = None
+        self.helper.stop_ensuring_object_permanence()
+        self._mission_complete(self.current_mission)
+        self.current_mission = None
+        self.current_mission_timeout = None
+
+    @util.cancellableInlineCallbacks
+    def _run_mission(self, mission):
+        self.pub_msn_info.publish(String("Starting Mission {}".format(mission.name)))
+        yield self.nh.sleep(.3)
+        self.running_mission = True
+        self.current_mission_name = mission.name
+        self.current_mission = mission
+        self.helper.ensure_object_permanence(mission.item_dep, self._object_gone_missing)
+        self.mission_defer = self.do_mission(mission)
+        self.current_mission_timeout = genpy.Duration(mission.timeout)
+        self.current_mission_start_time = self.nh.get_time()
+        self.mission_defer.addCallbacks(self._end_mission, errback=self._err_mission)
+
+    def _run_base_mission(self):
+        if self.base_mission is not None:
+            self.running_mission = True
+            self.current_mission_name = self.base_mission.name
+            self.mission_defer = self.do_mission(self.base_mission)
+            self.mission_defer.addCallbacks(self._end_base_mission, errback=self._err_base_mission)
+
+    @util.cancellableInlineCallbacks
+    def empty_queue(self):
+        """Constantly empties the queue if there is something in it, or run the base mission otherwise."""
+        starting_time = self.nh.get_time()
+        while self.keep_running:
+            if self.current_mission_timeout is not None:
+                if (self.nh.get_time() - self.current_mission_start_time) > self.current_mission_timeout:
+                    fprint(self.current_mission_name, msg_color="red", title="MISSION TIMEOUT:")
+                    self.mission_defer.cancel()
+            if self.running_mission:
+                self.refresh()
+                yield self.nh.sleep(1)
+                continue
+            try:
+                mission = self.queue.get(block=False)
+                yield self._run_mission(mission)
+            except que.Empty:
+                self._run_base_mission()
+            except Exception as exp:
+                fprint(exp, msg_color="red", title="LOL WHAT THE FUCK HAPPENED? THIS IS ON YOU TESS!!!")
```

Comment:
Even though I do it too, it's probably best not to curse in github code since it's all up here on the internet